python/cudf/cudf/core/buffer.py

```diff
@@ -153,3 +152,24 @@ def confirm_1d_contiguous(array_interface):
     if strides is not None:
         if not get_c_contiguity(shape, strides, itemsize):
             raise ValueError("Buffer data must be 1D C-contiguous")
+
+
+def get_c_contiguity(shape, strides, itemsize):
+    """
+    Determine if combination of array parameters represents a
+    c-contiguous array.
+    """
+    ndim = len(shape)
+
+    if ndim == 0 or (ndim == 1 and strides[0] == itemsize):
```

Comment:
Should also check if `strides is None` here:
```suggestion
    if ndim == 0 or strides is None or (ndim == 1 and strides[0] == itemsize):
```